### DISCUSSION TITLE
How to export react-table data into pdf?

### DISCUSSION THREAD

**Discussion by jasanidarshit** [TanStack/table#2513] [General]
Hey Community!
I have a dynamic react-table with expandable rows. I want to let users download the data however it is currently rendered in react-table using a button. 
Regular react-pdf or equivalent repositories don't work because the data is dynamic.

Thanks in advance!

**Comment by hmanaog** [comment]
Hi, I know this is 2 weeks old already, but I want to share how I've done mine, incase someone will look for this export to pdf again.

I was able to implement this with this PR from @gargroh : https://github.com/tannerlinsley/react-table/pull/1941
He created a nice react-table-plugin to export the react-table into CSV and Excel: https://github.com/gargroh/react-table-plugins
Check out the examples.

And for exporting to PDF i followed this one from stackoverflow and inserted it into the code react-table-plugin: https://stackoverflow.com/questions/56752113/export-to-pdf-in-react-table

I suggest you to check out the react-table-plugin and understand it, once you know how the react-table-plugin works, you'll be able to insert the export to PDF function.

Hope this helps.

**Comment by kumaryogesh4486** [comment]
dddddddddddddddd

**Comment by pszafer** [comment]
maybe just use browser print function with css styling
